src/Components/Web.JS/src/Virtualize.ts

```diff
@@ -0,0 +1,77 @@
+export const Virtualize = {
+  init,
+  dispose,
+};
+
+const observersByDotNetId = {};
+
+function findClosestScrollContainer(element: Element | null): Element | null {
+  if (!element) {
+    return null;
+  }
+
+  const style = getComputedStyle(element);
+
+  if (style.overflowY !== 'visible') {
+    return element;
+  }
+
+  return findClosestScrollContainer(element.parentElement);
+}
+
+function init(dotNetHelper: any, spacerAbove: Element, spacerBelow: Element, rootMargin = 50): void {
+  const intersectionObserver = new IntersectionObserver(intersectionCallback, {
+    root: findClosestScrollContainer(spacerAbove),
+    rootMargin: `${rootMargin}px`,
+  });
+
+  intersectionObserver.observe(spacerAbove);
+  intersectionObserver.observe(spacerBelow);
+
+  const mutationObserver = new MutationObserver((): void => {
+    intersectionObserver.unobserve(spacerAbove);
+    intersectionObserver.unobserve(spacerBelow);
+    intersectionObserver.observe(spacerAbove);
+    intersectionObserver.observe(spacerBelow);
```

Comment:
This code that unobserves then re-observes the same thing would be worthy of a comment to explain why we do this.

If I recall correctly, it's to allow for the case where even after adjusting the spacer sizes, one of them is still visible, so we want to immediately re-trigger the spacer size update logic.